web/config/settings/settings_base.py

```diff
@@ -235,6 +235,46 @@ class LoggerConfig(TypedDict, total=False):
     }
 }
 
+###
+### CELERY settings ###
+###
+
+CELERY_BROKER_URL = "redis://redis:6379/1"
+CELERY_ACCEPT_CONTENT = ["json"]
+CELERY_TASK_SERIALIZER = "json"
+CELERY_RESULT_SERIALIZER = "json"
+CELERY_TIMEZONE = "UTC"
+
+# These default time limits are useful in Perma, where capturing tasks
+# sometimes hang, for still-undiagnosed reasons, and where all tasks
+# are expected to be short-lived.
+#
+# It remains to be determined whether they will prove appropriate here.
+#
+# If a task is running longer than five minutes, ask it to shut down
+# CELERY_TASK_SOFT_TIME_LIMIT=300
+# If a task is running longer than seven minutes, kill it
+# CELERY_TASK_TIME_LIMIT = 420
+CELERY_TASK_TIME_LIMIT = 2
```

Comment:
I would start with whatever the default base time limit is, and use the soft default rather than the hard kill, so more reporting is potentially captured. I dunno that 2 seconds would tell us much of anything? Rendering a PDF for a 500 page book will definitely take longer than that!

Reply:
Ah, that was on purpose: I should have put my reasoning in a comment instead of the PR description. I set that to a ridiculously low number to forcibly remind us to think about and pick reasonable values when adding the actual PDF-making task.

[There is no default time limit](https://docs.celeryq.dev/en/stable/userguide/configuration.html#std-setting-task_time_limit); you have to set one explicitly. I definitely agree that we should set the "soft" limit to whatever we actually want the cap to be, and only use the hard limit to clean up clearly hanging tasks.... which might be less of an issue here than for Perma's capture process: I imagine there's less code that might potentially swallow the "soft" request to stop.

Happy to set some sensible values here instead if you like! What would you suggest? I haven't tried exporting any books yet.

Reply:
Ah sorry, I see the default is no-limit; that's actually consistent with other job runners like Airflow and Prefect so I should've assumed that.

I would typically set the global timeout starting with double the expected duration of the longest task. We haven't audited all the possible tasks, but if we include the ones currently implemented with cron:

* Site-wide search indexing - fast, just a few seconds
* Full-text searching - slow, ~10 minutes
* Reporting tables - fast, but could do more when run async
* PDF generation - seconds and probably not minutes, but not benchmarked

Playwright will have its own timeout since it's a browser initiating a request. How fast it runs will be very environment dependent. I have observed PagedJS failing to render very large books if it's deprived of RAM, but we'll want Playwright to handle its own timeout anyway so it can complete writing any screenshots or video files. 

I like to treat global task timeouts as a measure of last resort to stop literally stuck jobs, and use reporting tools to evaluate whether task duration is creeping up or otherwise seems unhealthy. If one particular task has the characteristics of both being typically very short and safe to kill (an example might be some kind of notification counter—a nice-to-have that is okay if it's a bit stale) then I might set a per-task timeout that's short assuming it'll just catch up on a subsequent run.

I find it's better to err on the side of longer timeouts, that give infrequently-long tasks room to complete, than very short timeouts that repeatedly kill jobs that can't possibly finish until the max is increased.

Reply:
Switched to a default soft time out of 20 minutes (twice the ~10min mentioned for full-text indexing) and a hard time out of 22 minutes.